mergify_cli/ci/junit_upload.py

```diff
@@ -0,0 +1,76 @@
+import pathlib
+import typing
+
+import httpx
+
+from mergify_cli import console
+from mergify_cli import utils
+
+
+def get_files_to_upload(
+    files: tuple[str, ...],
+) -> list[tuple[str, tuple[str, typing.BinaryIO, str]]]:
+    files_to_upload: list[tuple[str, tuple[str, typing.BinaryIO, str]]] = []
+
+    for file in set(files):
+        file_path = pathlib.Path(file)
+        files_to_upload.append(
+            ("files", (file_path.name, file_path.open("rb"), "application/xml")),
```

Comment:
I checked httpx code and it does not take care of closing the files, so you should do this at some point.
Ideally this function should be a context manager to do this properly.